extensions/positron-r/src/tasks.ts

```diff
@@ -67,17 +67,37 @@ export async function getRPackageTasks(editorFilePath?: string): Promise<vscode.
 		env['RSTUDIO_PANDOC'] = pandocPath;
 	}
 
-	// the explicit quoting treatment is necessary to avoid headaches on Windows, with PowerShell
-	return taskData.map(data => new vscode.Task(
-		{ type: 'rPackageTask', task: data.task, pkg: data.package },
-		vscode.TaskScope.Workspace,
-		data.message,
-		'R',
-		new vscode.ShellExecution(
-			binpath,
-			['-e', { value: data.rcode, quoting: vscode.ShellQuoting.Strong }],
-			{ env }
-		),
-		[]
-	));
+	return taskData.map(data => {
+		let exec: vscode.ProcessExecution | vscode.ShellExecution;
+		if (data.task === 'r.task.rmarkdownRender' && os.platform() === 'win32') {
+			// Using vscode.ProcessExecution gets around some hairy quoting issues on Windows,
+			// specifically encountered with PowerShell.
+			// https://github.com/posit-dev/positron/issues/3816
+			// We don't know of specific problems around not using a shell (for example, env vars
+			// appear to be inherited by ProcessExecution), but we're still scoping this narrowly
+			// out of caution.
```

Comment:
I created a new environment variable in `.zshrc` (this is macOS) and have confirmed that this env var is available to R code executed via `vscode.ProcessExecution`. Literally, I temporarily replaced the R snippet for `r.task.rmarkdownRender` with `Sys.getenv("FOOFY")` and, indeed, the env var is defined.

This behaviour is what you might expect given this documentation for `ProcessExecutionOptions`:

https://code.visualstudio.com/api/references/vscode-api#ProcessExecutionOptions

For `env`:

> The additional environment of the executed program or shell. If omitted the parent process' environment is used. If provided it is merged with the parent process' environment.

I think we are likely being overcautious here, which is fine. But if we find more places where `ProcessExectuion` looks attractive, I'm optimistic we could expand its use.

Reply:
Could we solve #2723 by moving to `ProcessExecution`? Should all of these really be `ProcessExecution`?

Reply:
No this doesn't directly help with #2723, which is about propagating env vars from the Console R session to the R code we're running "elsewhere".

> Should all of these really be `ProcessExecution`?

Maybe? Not clear at the moment.